worker/docker.go

```diff
@@ -275,6 +285,8 @@ func (m *DockerManager) createContainer(ctx context.Context, pipeline string, mo
 	m.containers[containerName] = rc
 	m.gpuContainers[gpu] = containerName
 
+	go m.watchContainer(ctx, rc)
```

Comment:
@victorges I like your changes! However, I think there’s an advantage to keeping containers warm for a certain amount of time for batch jobs if requests continue to come in. This could help improve response times and resource efficiency.

Reply:
I see! So we should not remove the container when the context is cancelled right? I can change the logic to just return the container on that case instead WDYT? With that I can even remove the "return container" function that is currently called explicitly, and instead just let the callers pass a ctx that is cancelled when the request is done and the container is auto-returned.

Reply:
Implemented on https://github.com/livepeer/ai-worker/pull/275/commits/4d5adbef48f81fc60489daf24367d8e452319dde !